rfd/0077-support-dump.md

```diff
@@ -0,0 +1,252 @@
+and customers requesting help. The Teleport development team should also
+greatly benefit from a standardized set of information about a problem cluster.
+
+Without a support dump process, this increases the burden on both ends of the
+interaction as relevant information is requested and inspected. This adds
+unnecessary time to solution with each subsequent requests for information. 
+
+A standardized support dump would be able to provide a large amount of
+information in one step, increasing visibility into the possible issues and
+decreasing time to solution for a given request. It will also greatly decrease
+the effort required to reproduce a particular issue, especially when there is a
+less common configuration in play.
+
+## Details
+
+### End User Experience
+
+Because teleport has many deployment methods, the support dump procedure should
+be smart enough to detect and gather information in the correct way. This may
+include an entire cluster, or a single agent. It should also be able to gather
+logs from standard locations wherever possible. It should advise the end user
+if it is unable to gather critical information for any reason.
+
+It should be built in to the Teleport binary itself. It should be possible to
+initiate a support dump from both the command line or from the web interface.
+Provisions should be made to advise about (or redact) possible sensitive
+information where applicable.
+
+File size limits will also need to be considered because the resulting files
+may be transmitted over systems with extremely limited file sizes.
+
+The end user should be able to run one command or click on one web gui button
+to generate a dump file. This file will likely need to be a compressed tar file.
+
+### Contents of the Dump File
+
+* The resulting compressed tar file should have a defined directory structure.
+* It should be both user-readable and machine-readable.
+* It should have timestamped log output.
+* When created for an entire cluster, it should be gathering information from every teleport agent/node/service that it is connected to the cluster.
+* It should include relevant objects such as users and roles to help troubleshoot rbac options.
+* any redactions should be done in a one way consistent obfuscation so that objects can still be uniquely differentiated.
+
+### Structure
+
+The top level `nodes/` directory will have basic information about each node in
+the cluster. If the node where the support dump is initiated, then it will
+include information about only that agent, plus information about the auth
+and/or proxy nodes it is aware of, along with tunnel status.
+
+The `objects/` directory will gather any objects it possibly can. If a support
+dump is initiated remotely with tsh credentials, it should indicate which
+objects it was unabled to gather due to permission issues. This should collect
+every object type it can possibly retrieve. The separate .error files would
```

Comment:
If I understand correctly, the motivation for this feature is to help support customers who are having an issue and ultimately get to resolution faster.

Collecting anything and everything that can "possibly be retrieved" does not necessarily lead to faster time to resolution. On very large clusters, for example, you may find that this makes the dump take longer to execute, decreases performance of the cluster while it is being collected, and provides the support engineer with lots of useless information to sift through.

I would rather start by identifying the things we most commonly ask for, collecting those, and then expanding scope if we discover more things should be included.